server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java

```diff
@@ -1214,10 +1255,83 @@ private void getObjectWithRange(final HttpServletResponse response, final Range
   private static String filenameFrom(final @PathVariable String bucketName,
       final HttpServletRequest request) {
     final String requestUri = request.getRequestURI();
-    return UrlEncoded.decodeString(
+    return objectNameToFileName(
+        UrlEncoded.decodeString(
             requestUri.substring(requestUri.indexOf(bucketName) + bucketName.length() + 1)
+        )
     );
   }
+  
+  /** 
+   * Escape object names (and prefixes) so that they can be safely mapped to a file name 
+   * as consumed by a {@link FileStore}. The encoding should work on at least Unix, Windows
+   * and macOS.
+   * 
+   * <p>The escaping is based on a modified URL encoding scheme (using 16 instead of 32 bits) 
+   * and uses different rules which characters to escape.
+   * 
+   * @param objectName the object name to encode
+   * @return encoded key
+   */
+  // @VisibleForTesting
+  static String objectNameToFileName(String objectName) {
```

Comment:
The two conversion methods here are good candidates to be extracted into other classes. 

I'll merge the PR anyway, as we've neglected this one for too long, but if you are still up to it, feel free to refactor a bit in a new PR.. ;)

Reply:
Can do - that would give me the opportunity to correct the method comment for `objectNameToFileName` whhich should have contained "using 16 instead of 8 bits", of course.
What were you thinking of, a separate utility class below `com.adobe.testing.s3mock.util`?